app/uk/gov/hmrc/gform/gform/csv/CsvTaxRateAdapter.scala

```diff
@@ -33,36 +38,33 @@ class CsvTaxRateAdapter extends CsvDataRetrieveAdapter[TaxRate] {
       TaxRate(
         row("TaxRegime"),
         row("Code"),
-        row("Rate"),
+        BigDecimal(row("Rate")).setScale(2),
         LocalDate.parse(row("StartDate"), hmrcTaxRateDateFormat),
         LocalDate.parse(row("EndDate"), hmrcTaxRateDateFormat)
       )
     }
 
-  override def search(request: DataRetrieve.Request): Option[JsValue] = {
-    val taxRateRequest: TaxRateRequest = request.json.as[TaxRateRequest]
-    val regimeRates: Option[List[TaxRate]] = data.groupBy(_.regime).get(taxRateRequest.regime)
-    val requestedDate: LocalDate = LocalDate.parse(taxRateRequest.date, hmrcTaxRateRequestFormat)
-
-    val searchResult: Option[List[TaxRate]] = regimeRates.map(_.filter { rate =>
-      rate.code == taxRateRequest.code && (
-        rate.startDate.isEqual(requestedDate) || rate.startDate.isBefore(requestedDate)
-      ) && (
-        rate.endDate.isEqual(requestedDate) || rate.endDate.isAfter(requestedDate)
-      )
-    })
+  override def search(request: DataRetrieve.Request): Option[JsValue] =
+    Try(request.json.as[TaxRateRequest]) match {
+      case Success(taxRateRequest: TaxRateRequest) =>
```

Comment:
We should use `Try` only if there is no better alternative. In this case we can use `validate` [method](https://github.com/playframework/play-json/blob/main/play-json/shared/src/main/scala/play/api/libs/json/JsReadable.scala#L37)

```
request.json.validate[TaxRateRequest] match {
      case JsSuccess(taxRateRequest, _) =>
```